Lib/test/test_asyncio/test_selector_events.py

```diff
@@ -363,9 +363,12 @@ def test_accept_connection_multiple(self):
         sock.accept.return_value = (mock.Mock(), mock.Mock())
         backlog = 100
         # Mock the coroutine generation for a connection to prevent
-        # warnings related to un-awaited coroutines.
+        # warnings related to un-awaited coroutines. _accept_connection2
+        # is an async function that is patched with AsyncMock. Use MagicMock
+        # since the coroutine is not awaited.
         mock_obj = mock.patch.object
-        with mock_obj(self.loop, '_accept_connection2') as accept2_mock:
+        with mock_obj(self.loop, '_accept_connection2',
+                      new=mock.MagicMock()) as accept2_mock:
```

Comment:
Interesting! I usually use new for created objects and `new_callable` when I want a different mock type.

Reply:
This change in behavior has to be documented as noted in the tracker. Currently there is below text where MagicMock is documented as the target type but that changed in 3.8 with `AsyncMock` for async functions and `MagicMock` for normal functions when new is omitted in mock.patch.

https://docs.python.org/3/library/unittest.mock.html#patch

> If new is omitted, then the target is replaced with a MagicMock